anxcloud/resource_dns_record.go

```diff
@@ -0,0 +1,225 @@
+package anxcloud
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/api/types"
+	clouddnsv1 "go.anx.io/go-anxcloud/pkg/apis/clouddns/v1"
+	"go.anx.io/go-anxcloud/pkg/utils/object/compare"
+)
+
+var syncDNSRecordOps sync.Mutex
+var dnsRecordSkipMutexLock = providerContextKey("dns-record-skip-mutex-lock")
+
+func resourceDNSRecord() *schema.Resource {
+	return &schema.Resource{
+		Description:   "This resource allows you to create DNS records for a specified zone. Records of type TXT might be unstable for now.",
```

Comment:
"TXT records might behave funny, we are working on it."

But probably just a preference thing - feel free to use or ignore :)